### ISSUE TITLE
support for unix-dgram

### ISSUE THREAD

**Issue by jgrammen-agilitypr** [bodgit/puppet-postfix#25]
The default postlog line in master.cf on ubuntu 20.04 postfix (3.4.13-0ubuntu1 ), contains the following
```
postlog   unix-dgram n  -       n       -       1       postlogd
```
the module errors on the following configuration
```
    postfix::master { 'postlog/unix-dgram':
      private => 'n',
      chroot  => 'n',
      limit   => '1',
      command => 'postlogd',
    }
```
throwing the following error
```
Error: Could not retrieve catalog from remote server: Error 500 on SERVER: Server Error: Evaluation Error: Error while evaluating a Resource Statement, Postfix::Master[postlog/unix-dgram]: parameter 'service' expects a match for Pattern[/(?x) ^ [^\/]+ \/ (?:inet|unix|fifo|pass) $/], got 'postlog/unix-dgram' (file: /etc/puppetlabs/code/modules/mm/manifests/nms/icinga.pp, line: 1134) on node ca04vlnms01.agilitypr.internal
```
It appears that the module does not support having unix-dgram as the service type

**Comment by jgrammen-agilitypr** [comment]
After re reading the documentation it appears that postfix::master should support unix-dgram
but when using the below configuration
```
    postfix::master { 'postlog':
      service => 'postlog',
      type    => 'unix-dgram',
      private => 'n',
      chroot  => 'n',
      limit   => '1',
      command => 'postlogd',
    }
```
i get the following error
```
Error: Could not retrieve catalog from remote server: Error 500 on SERVER: Server Error: Evaluation Error: Error while evaluating a Resource Statement, Postfix::Master[postlog]:
  has no parameter named 'type'
  parameter 'service' expects a match for Pattern[/(?x) ^ [^\/]+ \/ (?:inet|unix|fifo|pass) $/], got 'postlog' (file: /etc/puppetlabs/code/modules/mm/manifests/nms/icinga.pp, line: 1134) on node ca04vlnms01.agilitypr.internal
Warning: Not using cache on failed catalog
Error: Could not retrieve catalog; skipping run
```
Version 3.1.0 of the module

**Comment by bodgit** [comment]
There is no `type` parameter, the `service` parameter should be set to `postlog/unix-dgram`, but that will also use the `$title` by default so you can just use:
```puppet
    postfix::master { 'postlog/unix-dgram':
      private => 'n',
      chroot  => 'n',
      limit   => '1',
      command => 'postlogd',
    }
```
At least you could if I had remembered to allow `unix-dgram` as a valid type. I forgot to update the `postfix::master` define.

**Comment by bodgit** [comment]
This should be fixed now in v3.1.1 which should be now uploaded to the forge shortly.

Thanks for the bug report!

**Comment by jgrammen-agilitypr** [comment]
thank you for fixing it so quickly.
